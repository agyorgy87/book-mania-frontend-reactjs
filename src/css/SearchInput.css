
.search-input-container {
    width: 60% !important;
}

.bi {
    color: rgb(138, 138, 138);
    font-size: 20px;
}

.search-button {
    border-left: 1px solid #ced4da !important;
    border-top: 1px solid #ced4da !important;
    border-bottom: 1px solid #ced4da !important;
    border-top-left-radius: 80px !important;
    border-bottom-left-radius: 80px !important;
    background-color: #ffffff !important;
    height: 50px;
}

.search-button:focus {
    border: 2px solid #1e7b85 !important;
}

.search-input {
    border-top-right-radius: 80px !important;
    border-bottom-right-radius: 80px !important;
    font-size: 20px !important;
    background-color: #ffffff !important;
    height: 50px;
}

.search-input:focus {
    border-left: none !important;
    box-shadow: none !important;
    outline: none !important;
    border-color: #ced4da !important;
}

/*
[type="search"]::-webkit-search-cancel-button,
[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
  appearance: none;
}
*/