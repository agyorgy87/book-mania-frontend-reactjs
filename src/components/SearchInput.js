import '../css/SearchInput.css';
import React, { useState } from 'react';
import axios from "axios";

const SearchInput = ({setVisibleBooks, setShowResult, setSearchResult, setGenreSelect, setPriceSelect, setReleaseDateSelect, setSpecialSearchSelect, setPublisherSelect}) => { //{onChange}

    const envAndLocal = "http://localhost:3001" || process.env.REACT_APP_API_URL

    const [searchInput, setSearchInput] = useState("");

    const callAllTitlesAndAuthors = (e) => {
        if (e.trim() !== "") {
            axios.get(envAndLocal + `/get-book-title/${e}`)
                .then((response) => {
                    setVisibleBooks(response.data);
                })
                .catch((error) => {
                    //console.error("Error fetching book titles:", error);
                });
            setShowResult(true);
            setSearchResult(e);
            setGenreSelect("");
            setPriceSelect("");
            setReleaseDateSelect("");
            setSpecialSearchSelect("");
            setPublisherSelect("");
            setSearchInput(e.target.value);
        } else {
            axios.get(envAndLocal + "/get-all-books")
             .then((response) => {
                setVisibleBooks(response.data);
            });
            setShowResult(false);
            setSearchResult("");
        }
    }

    const handleInputChange = (e) => {
        const value = e.target.value;
        setSearchInput(value); 
        callAllTitlesAndAuthors(value);
    };

    const handleInputBlur = () => {
        setSearchInput(""); 
    };

    return (
        <div className="input-group mt-3 search-input-container">
            <button className="btn btn-lg search-button" type="button">
                <i className="bi bi-search"></i>
            </button>
                <input 
                    className="form-control search-input" 
                    type="search" 
                    id="example-search-input"
                    value={searchInput}
                    onChange={handleInputChange}
                    onBlur={handleInputBlur}
                    placeholder="Search"
                />                           
        </div> 
    )
}

export default SearchInput;