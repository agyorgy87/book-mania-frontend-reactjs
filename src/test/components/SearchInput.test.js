import renderer from 'react-test-renderer';
import SearchInput from '../../components/SearchInput';

test("test with function", () => {
    let tree = renderer
    .create(<SearchInput onChange={() => {}}/>)
    .toJSON()
    expect(tree).toMatchSnapshot();
});