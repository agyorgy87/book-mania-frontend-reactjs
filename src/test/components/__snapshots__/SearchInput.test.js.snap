// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with function 1`] = `
<div
  className="input-group mt-3 search-input-container"
>
  <button
    className="btn btn-lg search-button"
    type="button"
  >
    <i
      className="bi bi-search"
    />
  </button>
  <input
    className="form-control search-input"
    id="example-search-input"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
  />
</div>
`;
